### PR TITLE
[94X] L1 no hack conditions - cleaned

### PR DESCRIPTION
**Pull request by rekovic**
Configure emulators from DB.
This is a cleaned-up version of #20577.

Details:
- All but EMTF emulators configure from GlobalTag
- EMTF configures from DB, but not from GT. Configuration is set to 2017_MC.
- Includes needed fixes in addOn tests.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
the last commit of Reverting was counter productive...I forgot that  the change in L1REPACK:Full2015 it was needed for AddOn test which failed above.  Will revert the last commit.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
@davidlange6 
The differences in DQM comparisons observed in #20577 were understood and expected.

**Comment by rekovic** [comment]
@Dr15Jones, @vazzolini, @kmaeshima, @dmitrijus, @franzoni, @vanbesien
As this is the cleaned up version of a previously-signed #20577, please sign.

**Comment by davidlange6** [comment]
could you explain briefly? Its always best to separate technical and physics changes when possible...

**Comment by Dr15Jones** [comment]
+1
for  Utilities/ReleaseScripts

**Comment by dmitrijus** [comment]
+1

**Comment by rekovic** [comment]
@davidlange6

Differences in Muons BMTF are to differences in params available in external LUTs  vs Prototype payload used when configuring via GT.  The payload is updated and requested from AlCa to include GT.

Differences in Calo are due to static configuration which used the latest greatest configuration for all data IOVs,  while this PR choses the configuration according to the run numbers and IOV, which is the correct thing to do.

> could you explain briefly? Its always best to separate technical and physics changes when possible...
…
 On Oct 10, 2017, at 4:31 PM, rekovic ***@***.***> wrote:
 @davidlange6
 The differences in DQM comparisons observed in #20577 were understood and expected.

**Comment by davidlange6** [comment]
merge

**Comment by rekovic** [comment]
Test to see the effects of the further changes in  GTs

**Comment by rekovic** [comment]
please test with #20758

**Comment by rekovic** [comment]
probably need to open another PR for testing only.

**Comment by rekovic** [comment]
maybe not, tests just started.

File: L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc [hunk 1]
```diff
@@ -46,23 +47,27 @@
...
+#include "L1Trigger/L1TMuon/interface/L1TMuonGlobalParams_PUBLIC.h"
 
 #include "TMath.h"
 //
 // class declaration
 //
 using namespace l1t;
 
-  class L1TMuonProducer : public edm::stream::EDProducer<> {
+  class L1TMuonProducer : public edm::EDProducer {
```
Comment (thomreis) [inline L1Trigger/L1TMuon/plugins/L1TMuonProducer.cc]: @rekovic why was this moved back to being an EDProducer? Was this a mistake when trying to resolve the conflict?

**Comment by Martin-Grunewald** [comment]
Indeed - Can't have that for threading. Must be moved back to stream or so.
